torch/_inductor/codegen/triton.py

```diff
@@ -853,7 +860,7 @@ def should_use_persistent_reduction(self):
         if not (self.inside_reduction and config.triton.persistent_reductions):
             return False
         threshold = {
-            ReductionHint.INNER: 1024,
+            ReductionHint.INNER: 256,
```

Comment:
Note: the threshold for persistent kernel needs to be decreased to make sure that fox cases like `max(x).to(fp8)`:
1) XBLOCK size fulfill fp8 min_element_per_thread requirements. e.g. For fp8_e5m2 the min_element_per_thread is 4. In this case, XBLOCK size = 4 * NUM_WARPS (which is RBLOCK size / 128 by default) * 32 (warp_size)
2) XBLOCK * RBLOCK < 131072, which is Triton maximum tensor numel.
So XBLOCK * RBLOCK = RBLOCK_SIZE * RBLOCK_SIZE < 131072, max RBLOCK_size is 256.

There are lots of things can be tuned potentially. e.g. 
1) We may only update this rule when fp8 conversion is followed by a reduction. This would need some code refactoring as it seems no way to get this information without actually running `ops.to_dtype()`.
2) We could also reduce NUM_WARPS. However this would reduce parallelism, which doesn't seem to be ideal.
3) For a normal reduction kernel (i.e. not a persistent reduction kernel), we may also want to decrease split-reduction threshold so that each block handles fewer number of elements.